### PR TITLE
Remove redundant configs for Capybara

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

There are some configs for Capybara that I believe are not needed - see comments inline.

**Which issue(s) this PR fixes** 

**Verification steps** 

Make sure all tests works :) 

**Special notes for your reviewer**:

### CODE REVIEW

File: features/support/capybara.rb [hunk 1]
```diff
@@ -13,37 +13,17 @@
...
-
-Before '@chrome' do
-  Capybara.current_driver = :chrome
-end
-
-Before '@firefox' do
-  Capybara.current_driver = :firefox
+  # Capybara 3 changes the default server to Puma. It can be reverted to the previous default of WEBRick by specifying:
+  config.server = :webrick
+  config.disable_animation = true
```
Comment (mayorova) [inline features/support/capybara.rb:26]: This `disable_animation` is the only one that is not documented here: https://rubydoc.info/github/jnicklas/capybara/Capybara#configure-class_method

But the tests pass, so either it works fine, or this setting is irrelevant.

File: features/support/capybara.rb [hunk 2]
```diff
@@ -13,37 +13,17 @@
 # in case firefox is needed!
 # DEFAULT_JS_DRIVER = :headless_firefox
 
-Capybara.default_driver = :rack_test
-Capybara.javascript_driver = DEFAULT_JS_DRIVER
-Capybara.default_selector    = :css
-Capybara.disable_animation = true
-
-# Capybara 3 changes the default server to Puma. It can be reverted to the previous default of WEBRick by specifying:
-Capybara.server = :webrick
```
[the remark this answers is not in the document]
Comment (akostadinov) [reply features/support/capybara.rb:22]: But I don't see webrick set anywhere else. Although I'm not sure why do we need to use it, given in both - production and dev we use unicorn and puma...

**Review by akostadinov — APPROVED** [review]
looks nice! good to check if there are useful ideas also in #3911

**Comment by mayorova** [comment]
> looks nice! good to check if there are useful ideas also in #3911

Oh wow! I missed that one completely :facepalm: 

Thanks, I'll take a look.

**Review by jlledom — DISMISSED** [review]

File: features/step_definitions/provider_steps.rb [hunk 3]
```diff
@@ -237,6 +237,8 @@ def setup_provider(login)
   set_current_domain(@provider.external_admin_domain)
   stub_integration_errors_dashboard
 
+  @provider.users.first.user_sessions.create! # Prevents welcome flash from showing up
```
Comment (jlledom) [inline features/step_definitions/provider_steps.rb:240]: I understand the reason, but doesn't this make the `login` param useless? it will create a session anyway... and doesn't this break any test?

File: features/step_definitions/web_steps.rb [hunk 4]
```diff
@@ -143,14 +143,10 @@ def with_scope(locator)
...
-  if rack_test?
+  accept_confirm(text) do
     step original
-  else
-    accept_confirm(text) do
-      step original
-    end
-    wait_for_requests
   end
+  wait_for_requests
```
Comment (jlledom) [inline features/step_definitions/web_steps.rb:149]: I need some help to understand this

Comment (mayorova) [reply features/step_definitions/web_steps.rb:149]: I don't quite understand it either, I just copy-pasted it from @josemigallas 's PR :)

File: features/step_definitions/pagination_steps.rb [hunk 5]
```diff
@@ -22,7 +22,8 @@
       click_link page
     end
   else
-    input = find('input[aria-label="Current page"]')
+    # Pagination controls are rendered twice: at the top and at the bottom of the data table
+    input = find('input[aria-label="Current page"]', match: :first)
```
Comment (mayorova) [inline features/step_definitions/pagination_steps.rb:26]: This change is not really needed anymore, it was needed when I tried `config.match = :smart` match mode for Cabybara.

But probably we can leave it here to state that we are aware of this duplication, and have it fixed, if we ever switch to `:smart` mode again. (currently there were too many changes to perform here)

File: features/support/capybara.rb [hunk 6]
```diff
@@ -1,117 +1,48 @@
...
-Capybara.disable_animation = true
 
-# Capybara 3 changes the default server to Puma. It can be reverted to the previous default of WEBRick by specifying:
-Capybara.server = :webrick
-
-# see http://www.elabs.se/blog/60-introducing-capybara-2-1
 Capybara.configure do |config|
   config.default_driver = :rack_test
-  config.javascript_driver = DEFAULT_JS_DRIVER
+  config.default_selector = :css
```
Comment (mayorova) [inline features/support/capybara.rb:9]: These first three lines are actually the defaults, so they could be removed potentially. I decided to keep them, because I think it's clearer if they are explicit and not hidden within Capybara's code (or docs).

But if anybody thinks they are redundant, we can drop them.

File: features/step_definitions/cms/sidebar_steps.rb [hunk 7]
```diff
@@ -5,8 +5,6 @@
 end
 
 When /^I switch to (builtin|3scale) content$/ do |group|
-  ensure_javascript
```
Comment (akostadinov) [inline features/step_definitions/cms/sidebar_steps.rb:8]: why is this removed?

File: features/step_definitions/web_steps.rb (hunk 4 above)
Comment (akostadinov) [reply features/step_definitions/web_steps.rb:149]: maybe internally `accept_confirm` does nothing in rack_test so no need to complicate the code here, just a guess

File: features/support/capybara.rb [hunk 8]
```diff
@@ -1,117 +1,48 @@
...
-  Capybara.current_driver = :chrome
-end
-
-Before '@firefox' do
-  Capybara.current_driver = :firefox
-end
-
-Around '@security' do |scenario, block|
-  with_forgery_protection(&block)
-end
```
Comment (akostadinov) [inline features/support/capybara.rb:51]: not used?

File: features/support/capybara.rb [hunk 9]
```diff
@@ -1,117 +1,48 @@
...
-require 'cucumber/formatter/interceptor'
-class Cucumber::Formatter::Interceptor::Pipe
-  def is_a?(klass)
-    super || klass == IO
-  end
-end
-
-Capybara.register_driver :firefox do |app|
-  Capybara::Selenium::Driver.new(app, browser: :firefox)
-end
```
Comment (akostadinov) [inline features/support/capybara.rb:67]: we wont be able to run with firefox?

File: features/support/capybara.rb (hunk 8 above)
Comment (akostadinov) [reply features/support/capybara.rb:51]: moved, I see, thanks

File: features/support/capybara.rb (hunk 9 above)
Comment (mayorova) [reply features/support/capybara.rb:67]: Apparently, not :grimacing: 
Again, I just applied Josemi's suggestion. 

I don't think I have run automated tests in firefox myself, but if you have and want to bring it back, I'll do it.

Comment (akostadinov) [reply features/support/capybara.rb:67]: Yeah, I think it is nice to be able to run something with it in case we see strange results with chrome. This is the only browser we ship with RHEL. And Chrome is evil.

File: features/step_definitions/cms/sidebar_steps.rb (hunk 7 above)
Comment (akostadinov) [reply features/step_definitions/cms/sidebar_steps.rb:8]: I think it's nice to stay so that we know this is expected to fail without js. Otherwise it may require more cycles to figure out.

File: features/step_definitions/web_steps.rb (hunk 4 above)
Comment (mayorova) [reply features/step_definitions/web_steps.rb:149]: I think the thing here is that `rack_test` doesn't support modals, and it throws an exception when trying to call it: https://github.com/teamcapybara/capybara/blob/3.40.0/lib/capybara/driver/base.rb#L138-L140

But I think that this step was just simplified, because any test where you would expect to have a modal window (and thus where you would use this step) will need to be `@javascript` anyway. Nowadays, I think any admin portal tests MUST have `@javascript`, because even the login form is JS-based, and most of the menus etc.

So I think each time we have less non-JS tests, so this is why we are simplifying.

File: features/support/capybara.rb (hunk 9 above)
Comment (jlledom) [reply features/support/capybara.rb:67]: I agree Chrome is evil, in fact we should use only Firefox, but I guess clients use Chrome. IMO it's fine to keep both.

File: features/support/capybara.rb [hunk 10]
```diff
@@ -1,117 +1,59 @@
...
-  timeout = 120 # default 60
-  client = Selenium::WebDriver::Remote::Http::Default.new(open_timeout: timeout, read_timeout: timeout)
+  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options, timeout: 120)
+end
+
+# NOTE: depending on the scenario you will need to add both
+# @firefox/@headless_firefox AND @javascript tags
+Before '@firefox', '@headless_firefox' do
+  Capybara.javascript_driver = :headless_firefox
+end
```
Comment (jlledom) [inline features/support/capybara.rb:46]: Is this not redundant because of https://github.com/teamcapybara/capybara/blob/3.40.0/lib/capybara/cucumber.rb#L22-L27?

File: features/support/0001_world.rb [hunk 11]
```diff
@@ -4,7 +4,6 @@
 World(PlanHelpers)
 World(DummyAttachments)
 
-World(JsHelpers)
```
Comment (jlledom) [inline features/support/0001_world.rb:7]: What's this file for?

File: features/support/capybara.rb (hunk 6 above)
Comment (jlledom) [reply features/support/capybara.rb]: Did you finally remove them?

File: features/support/0001_world.rb (hunk 11 above)
Comment (mayorova) [reply features/support/0001_world.rb:7]: Looks like it just loads all the helpers. I was tempted to add this `World(Whatever)` in the same `whatever.rb` file, like `World(CapybaraExtensions)`

But then I thought - probably the order is important? :thinking: 
by the name of the file (starting with `0`), I guess it was done so to ensure the order of loading those helpers, although I am not sure whether it is needed. I decided not to change it here.

File: features/support/capybara.rb (hunk 10 above)
Comment (mayorova) [reply features/support/capybara.rb:46]: Not really.
That snippet from the link sets the `current_driver` to the driver with the name from the tag. It will actually also be executed each time `@firefox` or `@headless_firefox` are set.

Our code, however, sets the `javascript_driver` for **both** firefox tags, this is to override the `config.javascript_driver = :headless_chrome` set globally.

File: features/support/0001_world.rb (hunk 11 above)
Comment (jlledom) [reply features/support/0001_world.rb:7]: @akostadinov I guess both. Anyway it's clear now.
